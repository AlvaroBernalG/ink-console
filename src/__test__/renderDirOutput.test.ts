import {inspect} from 'util';
import renderDirOutput from '../renderDirOutput';

test('renderDirOutput', () => {
  expect(
    renderDirOutput({type: 'dir', value: 5}, {depth: 0, expandKey: 'l'}),
  ).toBe(inspect(5, {colors: true}));
  expect(
    renderDirOutput({type: 'dir', value: 'foo'}, {depth: 0, expandKey: 'l'}),
  ).toBe(inspect('foo', {colors: true}));
  expect(
    renderDirOutput({type: 'dir', value: {}}, {depth: 0, expandKey: 'l'}),
  ).toMatchSnapshot('{}, depth=0');
  expect(
    renderDirOutput({type: 'dir', value: {}}, {depth: 1, expandKey: 'l'}),
  ).toMatchSnapshot('{}, depth=1');
  expect(
    renderDirOutput({type: 'dir', value: {v: {}}}, {depth: 1, expandKey: 'l'}),
  ).toMatchSnapshot('{v: {}}, depth=1');
  expect(
    renderDirOutput({type: 'dir', value: {v: {}}}, {depth: 2, expandKey: 'l'}),
  ).toMatchSnapshot('{v: {}}, depth=2');
});
