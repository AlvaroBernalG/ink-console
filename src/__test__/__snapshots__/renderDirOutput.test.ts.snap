// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`{}, depth=0 1`] = `"{}"`;

exports[`{}, depth=1 1`] = `"{}"`;

exports[`{v: {}}, depth=1 1`] = `
"{
  v: {},
}"
`;

exports[`{v: {}}, depth=2 1`] = `
"{
  v: {},
}"
`;
